packages/playwright-core/src/server/fetch.ts

```diff
@@ -360,6 +360,7 @@ export abstract class APIRequestContext extends SdkObject {
             if (e)
               reject(new Error(`failed to decompress '${encoding}' encoding: ${e}`));
           });
+          body.on('error', e => reject(new Error(`failed to decompress '${encoding}' encoding: ${e}`)));
```

Comment:
This was required otherwise it ended up in an unhandledException:

```
 Error: unexpected end of file

        at BrotliDecoder.zlibOnError [as onerror] (node:zlib:189:17)
```